rc/base/auto-pairs.kak

```diff
@@ -0,0 +1,102 @@
+decl str-list auto_pairs %((,):{,}:[,]:<,>:",":',':`,`)
+
+def -hidden -params 2 auto-pairs-insert-opener %{ try %{
+  exec -draft ';<a-K>[[:alnum:]]<ret>'
+  exec -no-hooks "%arg{2}<a-;>H"
+}}
+
+def -hidden -params 2 auto-pairs-insert-closer %{ try %{
+  exec -draft ";<a-k>\Q%arg{2}<ret>d"
+}}
+
+def -hidden -params 2 auto-pairs-delete-opener %{ try %{
+  exec -draft ";<a-k>\Q%arg{2}<ret>d"
+}}
+
+def -hidden -params 2 auto-pairs-delete-closer %{ try %{
+  exec -draft "h<a-k>\Q%arg{1}<ret>d"
+}}
+
+def -hidden auto-pairs-insert-new-line %{ try %{
+  %sh{
+    regex=$(printf '\Q%s\E' "$kak_opt_auto_pairs" | sed s/:/'\\E|\\Q'/g';'s/'<,>'/'<lt>,<gt>'/g';'s/,/'\\E\\n\\h*\\Q'/g)
+    echo "exec -draft %(;KGl<a-k>$regex<ret>)"
```

Comment:
Since the `$regex` variable contains backslashes, passing this string to `echo` is risky, better use `printf`.